### PR TITLE
enh(log): exit log() earlier if no crashreport registered

### PR DESCRIPTION
**Pull request by ArtificialOwl**
follow-up on the previous improvement from @juliushaertl : https://github.com/nextcloud/server/pull/35970

The idea is to avoid some formatting of the payload as soon as we know for sure that it will not be used

### CODE REVIEW

**Comment by ArtificialOwl** [comment]
_(failing test not related)_

**Review by icewind1991 — CHANGES_REQUESTED** [review]
This would break `BeforeMessageLoggedEvent`

**Comment by ArtificialOwl** [comment]
> This would break `BeforeMessageLoggedEvent`

I added a `hasListeners()` and exit the method if no `BeforeMessageLoggedEvent` is registered. 
Not sure the performance gain anymore. 
Your thoughts ?

File: lib/private/Log.php [hunk 1]
```diff
@@ -329,8 +326,8 @@ public function logException(Throwable $exception, array $context = []) {
 		$level = $context['level'] ?? ILogger::ERROR;
 
 		$minLevel = $this->getLogLevel($context);
-		if ($level < $minLevel && ($this->crashReporters === null || !$this->crashReporters->hasReporters())) {
-			return;
+		if ($level < $minLevel && (($this->crashReporters?->hasReporters() ?? false) === false)) {
```
Comment (icewind1991) [inline lib/private/Log.php:329]: missing the `hasListeners` check here

Comment (ArtificialOwl) [reply lib/private/Log.php:329]: this fix a known issue ?

**Review by icewind1991 — APPROVED** [review]
log events still work

**Review by skjnldsv — APPROVED** [review]

**Pull request merged**